### DISCUSSION TITLE
How to configure concourse to be able to login using the fly command if I have an LDAP

### DISCUSSION THREAD

**Discussion by roDew** [concourse/concourse#6050] [Help & Support] [answered]
Hi

This thread is a follow up of #5726 and #3208 
Apparently, after the merge of [dexidp/dex#1621](https://github.com/dexidp/dex/pull/1621) concourse should be able to support login by CLI without interacting with browser when an LDAP is configured. 
If understood correctly, there should be a way to configure the PasswordConnector property to use ldap instead of local, but it seems I'm missing something as I cannot find how to do it via the common environmental variables that concourse uses (CONCOURSE_*).

Can anybody give me a hint with this?

Thank you very much in advance :)

**Comment by jamieklassen** [comment, marked as answer]
Unfortunately authenticating against an LDAP server with `fly login -u -p` is still currently impossible, regardless of how your concourse server is configured -- to get this behaviour, code changes to concourse are required. [This comment](https://github.com/concourse/concourse/issues/3208#issuecomment-642961415) describes the fact that the code changes required are not huge, but nobody has issued a successful PR yet. @SriharshaM made an attempt with #5796, but in order to support multiple password connectors a slightly more subtle change needs to be made. I gave some suggestions about how to achieve this [here](https://github.com/concourse/concourse/pull/5796#pullrequestreview-435233374). If you're interested in pursuing this I would _probably_ be the person reviewing any PRs and giving feedback.

**Reply by roDew** [reply]
Thank you very much for answering!
I will take a look at the required changes and see if I bring myself to do it :)

**Reply by ywei2017** [reply]
@roDew Have you got a chance to submit the PR and fix the issue?
Thanks
